CRM/Utils/Check/Component/OptionGroups.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2016                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2016
+ */
+class CRM_Utils_Check_Component_OptionGroups extends CRM_Utils_Check_Component {
+
+  /**
+   * @return array
+   */
+  public function checkOptionGroupValues() {
+    $messages = array();
+    $problemValues = array();
+    $optionGroups  = civicrm_api3('OptionGroup', 'get', array(
+      'sequential' => 1,
+      'data_type' => array('IS NOT NULL' => 1),
+    ));
+    if ($optionGroups['count'] > 0) {
+      foreach ($optionGroups['values'] as $optionGroup) {
+        $values = civicrm_api3('OptionValue', 'get', array(
+          'sequential' => 1,
+          'option_group_id' => $optionGroup['name'],
+        ));
```

Comment:
I think this will be only limited to a count of 25. Can we use a BAO function `getOptionValuesArray()` instead of an API call here too ?

    $values = CRM_Core_BAO_OptionValue::getOptionValuesArray(option_group_id);